TrkQual ANN1 v2

AndrewEdmonds11 opened the pull request:
This PR adds the retrained TrkQual ANN1 to Offline. ~Note that although the ANN structure didn't change, we still ended up with a different .hxx file. For this reason, I have added the ability for the ```TrackQuality``` module to use either the old or "new" models.~ The code to train is in Mu2e/MLTrain#1

Edit: after the meeting today, Dave pointed me to a script that reformats the .dat file so that we don't need two different .hxx files for the same model. I have removed this functionality from TrackQuality. The script to convert the .dat file is now in Mu2e/MLTrain#1

FNALbuild commented:
Hi @AndrewEdmonds11,
You have proposed changes to files in these packages:
- TrkDiag

which require these tests: build.

@Mu2e/fnalbuild-users, @Mu2e/write have access to CI actions on [main](/Mu2e/Offline/tree/main).





:hourglass: The following tests have been triggered for e43a2f0b053f998a29633c2b3bb8e542d3c869d8: build  (Build queue - API unavailable)


[About FNALbuild.](https://mu2ewiki.fnal.gov/wiki/GitHubWorkflow#GitHub_Pull_Request_Procedures_and_FNALbuild) [Code review on Mu2e/Offline.](https://mu2ewiki.fnal.gov/wiki/GitHubWorkflow#Code_Review)

FNALbuild commented:
:sunny: The build tests passed at e43a2f0b053f998a29633c2b3bb8e542d3c869d8.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| test with | :white_check_mark: | Command did not list any other PRs to include |
| merge | :white_check_mark: | Merged e43a2f0b053f998a29633c2b3bb8e542d3c869d8 at 8b5edb9bd36eb7cc42c63c1bb92c19eafeb947e4 |
| build (prof) | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2727/artifact/scons.log). Build time: 04 min 36 sec |
| ceSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2727/artifact/ceSimReco.log)  |
| g4test_03MT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2727/artifact/g4test_03MT.log)  |
| transportOnly | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2727/artifact/transportOnly.log)  |
| POT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2727/artifact/POT.log)  |
| g4study | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2727/artifact/g4study.log)  |
| cosmicSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2727/artifact/cosmicSimReco.log)  |
| cosmicOffSpill | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2727/artifact/cosmicOffSpill.log)  |
| ceSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2727/artifact/ceSteps.log)  |
| ceDigi | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2727/artifact/ceDigi.log)  |
| muDauSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2727/artifact/muDauSteps.log)  |
| ceMix | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2727/artifact/ceMix.log)  |
| rootOverlaps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2727/artifact/rootOverlaps.log)  |
| g4surfaceCheck | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2727/artifact/g4surfaceCheck.log)  |
| trigger | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2727/artifact/trigger.log)  |
| check_cmake | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2727/artifact/check_cmake.log)  |
| FIXME, TODO | :white_check_mark: | [TODO (0) FIXME (0) in 1 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2727/artifact/fixme_todo.log) |
| clang-tidy | :large_orange_diamond: | [2 errors 29 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2727/artifact/clang-tidy.log) |
| whitespace check | :white_check_mark: | [no whitespace errors found](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2727/artifact/whitespace_errs.log) |


N.B. These results were obtained from a build of this Pull Request at e43a2f0b053f998a29633c2b3bb8e542d3c869d8 after being merged into the base branch at 8b5edb9bd36eb7cc42c63c1bb92c19eafeb947e4.


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//2727/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.

brownd1978 reviewed (commented):
Andy, did the set of variables used change between v1 and v2? Or is it just that the training + SOPHIE produced a different data struct that forces a 2nd version?

AndrewEdmonds11 commented:
After the meeting today, Dave pointed me to a script that reformats the .dat file so that we don't need two different .hxx files for the same model. I have removed this functionality from TrackQuality. The script to convert the .dat file is now in https://github.com/Mu2e/MLTrain/pull/1. I have updated the lead comment of this PR to make this clear to anyone reading in the future

This PR is now ready to merge

AndrewEdmonds11 commented:
I forgot to add that I tested that the converted .dat with the old .hxx and the original .dat with the new .hxx produce the same results. In the plot below trkqual3 is the new .hxx with the unconverted .dat, and trkqual4 is the old .hxx with the converted .dat. The difference between all values is 0

<img width="1444" height="986" alt="image" src="https://github.com/user-attachments/assets/a322f4c7-99e8-4098-858c-28dbb5d12bfc" />

brownd1978 commented on TrkDiag/inc/TrkQual_ANN1_v2.hxx:
```diff
@@ -0,0 +1,262 @@
+//Code generated automatically by TMVA for Inference of Model file [TrkQual_ANN1_v2.onnx] at [Wed Sep 24 16:57:16 2025] 
```
Is this file still needed?

brownd1978 reviewed (commented):
I maybe misunderstood the effect of your last update WRT *v2.hxx
